src/org/traccar/protocol/TzoneProtocolDecoder.java

```diff
@@ -120,7 +120,7 @@ protected Object decode(
             return null;
         }
         int hardware = buf.readUnsignedShort(); // model
-        buf.readUnsignedInt(); // firmware
+        long firmware = buf.readUnsignedInt(); // firmware
```

Comment:
I need to put hardware and firmware into variables as this data is parsed prior other data and all prior to a Position variable being instantiated.